### ISSUE TITLE
Missed type error in subclass methods

### ISSUE THREAD

**Issue by adrianbn** [facebook/pyre-check#161]
Pyre (0.0.25) fails to report an incompatible parameter type error when a subclass calls a method defined in a superclass with the wrong parameter types. 

```python
class Container:
    def put(self, code: int) -> None:
        print(f"Type of code: {type(code)} with value {code}")

class Sub(Container):
    def __init__(self):
        self.put("This should be an int but it is not reported")
        self.put(200)

s = Sub()
c = Container()
c.put("This is reported")
```

Here's the output of running pyre on the above example:

```
> pyre --source-directory . check
 ƛ Setting up a `.pyre_configuration` with `pyre init` may reduce overhead.
 ƛ Found 1 type error!
test.py:13:6 Incompatible parameter type [6]: Expected `int` for 1st anonymous parameter to call `test.Container.put` but got `str`.
```

**Comment by grievejia** [comment]
Hi @adrianbn,

Thanks for the report! This is a known issue: we currently disable type checking on constructors (at the moment, enabling it would lead to excessive number of false alarms). The issue is currently being actively worked on and we hope it's going to be addressed in the near future.

**Comment by adrianbn** [comment]
Hey @grievejia,

it is not a constructor issue. I just used the constructor to make the example simpler. This also fails to report:

```python
class Container:
    def put(self, code: int) -> None:
        print(f"Type of code: {type(code)} with value {code}")

class Sub(Container):
    def __init__(self):
        pass

    def not_constructor(self):
        self.put(code="This should be an int")
        self.put(code=200)

s = Sub()
s.not_constructor()
c = Container()
c.put(code="This is reported")
```

**Comment by grievejia** [comment]
In your second example, method `not_constructor` is not annotated at all. By default, pyre suppress errors in functions that have no annotations on it. If you add, say, a return type to `not_constructor`, the type error would surface. You can also add `# pyre-strict` at the top of the file to enable strict mode, where pyre would complain about both the error and about not adding annotations.

**Comment by adrianbn** [comment]
Got it, thanks! I find it odd that the lack of return annotation on the method makes pyre ignore all of the code inside of it, even code that does not affect the return value of the method. 

I'll close this one since you mentioned the constructor issue is being tracked elsewhere.
